Marlin/Configuration.h

```diff
@@ -413,6 +422,10 @@ const bool Z_PROBE_ENDSTOP_INVERTING = false; // set to true to invert the logic
 
 #ifdef ENABLE_AUTO_BED_LEVELING
 
+  #ifdef MESH_BED_LEVELING
+    #error Select 'ENABLE_AUTO_BED_LEVELING' or 'MESH_BED_LEVELING', not both
+  #endif // MESH_BED_LEVELING
+
```

Comment:
These tests should really be moved to `SanityCheck.h`. There might be other `MESH_BED_LEVELING` stuff in there already.
```
#ifdef MESH_BED_LEVELING
  #ifdef DELTA
    #error MESH_BED_LEVELING does not yet support DELTA printers
  #endif
  #ifdef ENABLE_AUTO_BED_LEVELING
    #error Select ENABLE_AUTO_BED_LEVELING or MESH_BED_LEVELING, not both
  #endif
#endif
```